package-system/AWSGameLiftServerSDK/README.md

```diff
@@ -0,0 +1,117 @@
+   to make sure boost library get build with correct toolset (clang-6.0 for example) and variants
+   * Under `GameLift-Cpp-ServerSDK-{version-number}` directory, run command `git init` to init a local git repository with current directory
+   * Run command `git add .` to stage all existing files
+   * Run command `git commit -m "init commit"` to commit your existing files
+   * Run command `git apply --ignore-space-change --ignore-whitespace AWSGameLiftServerSDK_3.4.1.patch` to apply patch 
+   * Run command `git diff` to check expected changes like
+     ```diff
+          BUILD_COMMAND b2 address-model=${am} ${boost_with_args}
+        )
+      else()
+     +  if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+     +    set(_build_variant "debug")
+     +  else()
+     +    set(_build_variant  "release")
+     +  endif()
+     +
+        list(APPEND boost_with_args
+     -    "cxxflags=-fPIC")
+     +    "cxxflags=-fPIC" "toolset=clang-6.0" "variant=${_build_variant}")
+        set(boost_cmds
+          CONFIGURE_COMMAND ./bootstrap.sh --prefix=<INSTALL_DIR>
+          BUILD_COMMAND ./b2 address-model=${am} ${boost_with_args}
+     ```
+
+5. Run the python script `build_and_package.py` under `GameLift-Cpp-ServerSDK-{version-number}` directory, like
+   ```
+   & python3 build_and_package.py --config gamelift-sdk.json --platform linux --version 3.4.1
+   ```
+
+6. Build package will be located under `GameLift-Cpp-ServerSDK-{version-number}/build/{version-number}` directory
+
+7. Copy and put build package into `package-system/AWSGameLiftServerSDK/linux/AWSGameLiftServerSDK`, for example
+   ``` 
+   - AWSGameLiftServerSDK
+     - include
+     - lib
+     - LICENSE_AMAZON_GAMELIFT_SDK.txt
+     - NOTICE_C++_AMAZON_GAMELIFT_SDK.txt
+   ```
+8. Update `package-system/AWSGameLiftServerSDK/linux/PackageInfo.json` with built version `PackageName`
+
+9. Update `<3p-package-source path>/package_build_list_host_linux.json` with updated `PackageName` in step 6
+
+## FAQ
+* For windows build, use `--msbuild_path` argument to specify msbuild executable path if msbuild is unrecognized as a valid command, like
+  ```
+  & python build_and_package.py --config gamelift-sdk.json --platform windows --version 3.4.1 --msbuild_path "C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\15.0\Bin\MSBuild.exe"
+  ```
+
+* For windows build, if you get error message `The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters`
+  Please move `GameLift-Cpp-ServerSDK-{version-number}` directory out of nested directories and retry.
+
+* This SDK is known to work with these CMake generators:
+  * Visual Studio 15 2017 Win64
+  * Visual Studio 14 2015 Win64
+  * Visual Studio 12 2013 Win64
+  * Visual Studio 11 2012 Win64
+  * Unix MakeFiles
+    
+  By default, we use `vs2017` for windows build and `clang-6.0` for linux build.
```

Comment:
Chances are it will build with VS2019 with no issue, so we should just change to specify vs2019 as the default and remove previous versions from being supported